### ISSUE TITLE
Logout not working for SSO

### ISSUE THREAD

**Issue by TommyJ1994** [aerogear/aerogear-android-sdk#148]
## Expected Behavior
When I logout with SSO, all applications which are signed in will now show as no longer being authenticated and I will be informed when I enter another SSO app that the session has ended.

## Actual Behavior
Logout with SSO doesn't work. Even after killing both apps, one will ask me to login, the other will still show the users details.

## Steps to Reproduce the Problem

  1. Create two apps for SSO.
  1. Login in one, you will be authenticated automatically in the second.
  1. Logout in one of the apps. Check the other, the app shows as still authenticated. Kill the app and restart and you will still be shown as logged in.

## Specifications

  - Relevant Service Modules: Auth

**Comment by secondsun** [comment]
So I don't think this is a bug.  Apps aren't sharing their log in state with each other and until an app uses the expired token they won't know that they have been logged out.  It is a side effect of the way that OAuth works / is implemented.

If we want to support this use case then what we have to do is create an android Authenticator implementation which will act as a single point of truth for log in state for a single account shared among multiple apps.

**Comment by TommyJ1994** [comment]
Yeah, so logged this as a bug because we implemented the SSO feature and closed the user story already - but not all functionality works - as what is expected as only the auth action uses a shared state. I'll add the enhancement label too.
